launcher/ui/themes/CatPainter.cpp

```diff
@@ -22,12 +22,33 @@
 
 CatPainter::CatPainter(const QString& path, QObject* parent) : QObject(parent)
 {
-    m_image = QPixmap(path);
+    // Attempt to load as a movie
+    m_movie = new QMovie(path, QByteArray(), this);
+    if (m_movie->isValid()) {
+        // Start the animation if it's a valid movie file
+        connect(m_movie, &QMovie::frameChanged, this, &CatPainter::updateFrame);
+        m_movie->start();
+    } else {
+        // Otherwise, load it as a static image
+        delete m_movie;
+        m_movie = nullptr;
+
+        m_image = QPixmap(path);
+    }
+}
+
+CatPainter::~CatPainter()
+{
+    if (m_movie)
+        delete m_movie;
```

Comment:
QMovie is a QObject with `this` as its parent. So this is already done automatically.